lib/active_merchant/billing/gateways/xpay.rb

```diff
@@ -15,7 +15,7 @@ class XpayGateway < Gateway
 
       ENDPOINTS_MAPPING = {
         purchase: 'orders/2steps/payment',
-        authorize: 'orders/2steps/init',
+        preauth: 'orders/2steps/init',
```

Comment:
What's the reason behind calling this `preauth` instead of just `authorize` as it was?

Reply:
the endpoint do not correspond to an authorize transaction, according to documentation this is the first step to the purchase and authorize transactions which are described with some other fields included in the request body.